salt/utils/parsers.py

```diff
@@ -968,9 +968,17 @@ def _mixin_before_exit(self):
                     # Log error only when running salt-master as a root user.
                     # Otherwise this can be ignored, since salt-master is able to
                     # overwrite the PIDfile on the next start.
-                    if not os.getuid():
-                        logger.info('PIDfile could not be deleted: %s', six.text_type(self.config['pidfile']))
+                    def log_error():
```

Comment:
Instead of the overhead of a function call here, I think it would be better just to define the error string and then call logger.info in each case, IMHO.